### PR TITLE
update mongo sink adding ssl options

### PR DESCRIPTION
**Pull request by AlvaroVega**
- [x] Update Doc

https://mongodb.github.io/mongo-java-driver/3.6/driver/tutorials/ssl/

overcomes  https://github.com/telefonicaid/fiware-cygnus/pull/2385

### CODE REVIEW

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java [hunk 1]
```diff
@@ -592,19 +623,33 @@ private MongoDatabase getDatabase(String dbName) {
...
-                    client = new MongoClient(servers, credential, new MongoClientOptions.Builder().build());
+                    client = new MongoClient(servers, credential, new MongoClientOptions.Builder().
+                                             sslEnabled(sslEnabled).
+                                             sslInvalidHostNameAllowed(sslInvalidHostNameAllowed).
+                                             sslContext(sslContext).
+                                             build());
                 }
             } else {
-                client = new MongoClient(servers);
+                MongoClientOptions options = MongoClientOptions.builder()
```
Comment (mrutid) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java:647]: this branch seems extrange, at least "structurally" with respect the previous ones. Could be?

Comment (AlvaroVega) [reply cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/2386/commits/d0d92ceaa7202091f9b6f95d7b40296baf7b88dc

**Review by mrutid — APPROVED** [review]
LGTM

**Comment by fgalan** [comment]
Instead of adding individual fields to configure MongoDB, a more scalable soluction would be to removing them all and replace for a mongo_uri parameter.

Similar to what has been done [recently in Orion Context Broker](https://github.com/telefonicaid/fiware-orion/releases/tag/4.0.0)

> This version removes `-dbhost`, `-rplSet`, `-dbTimeout`, `-dbuser`, `-dbAuthMech`, `-dbAuthDb`, `-dbSSL` and `-dbDisableRetryWrites` CLI parameters along with associated env vars (deprecated in 3.12.0 version), so you have to use `-dbURI` (env var `ORION_MONGO_URI`) instead. Have a look to [this documentation section](https://github.com/telefonicaid/fiware-orion/blob/4.0.0/doc/manuals/deprecated.md#mapping-to-mongouri-from-old-cli-parameters) if you need help to migrate the configuration.

**Comment by AlvaroVega** [comment]
> Instead of adding individual fields to configure MongoDB, a more scalable soluction would be to removing them all and replace for a mongo_uri parameter.
> 
> Similar to what has been done [recently in Orion Context Broker](https://github.com/telefonicaid/fiware-orion/releases/tag/4.0.0)
> 
> > This version removes `-dbhost`, `-rplSet`, `-dbTimeout`, `-dbuser`, `-dbAuthMech`, `-dbAuthDb`, `-dbSSL` and `-dbDisableRetryWrites` CLI parameters along with associated env vars (deprecated in 3.12.0 version), so you have to use `-dbURI` (env var `ORION_MONGO_URI`) instead. Have a look to [this documentation section](https://github.com/telefonicaid/fiware-orion/blob/4.0.0/doc/manuals/deprecated.md#mapping-to-mongouri-from-old-cli-parameters) if you need help to migrate the configuration.

Yeah! It would be nice a great feature, replace MongoClientOptions to MongoClientUri

**Pull request merged**